File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
N = int(input())
A = list(map(int,sys.stdin.readline().split()))
A.sort()

M = int(input())
B = list(map(int,sys.stdin.readline().split()))

def binary_search(target,arr):
    # arr.sort()
    start = 0
    end = N-1
    while start<=end:
        mid = (start+end)//2
        if arr[mid] == target:
            return 1
        elif arr[mid] < target:
            start = mid+1
        else:
            end = mid-1
    return 0

for i in B:
    print(binary_search(i,A))

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int N, M;
int arr[1000010];

void BinarySearch(int *arr, int x)
{
	int start = 0, end = N-1, mid = (start + end) / 2;
	while (start <= end)
	{
		if (arr[mid] == x)
		{
			std::cout << "1" << "\n";
			return;
		}
		else if (x < arr[mid])
		{
			end = mid - 1;
		}
		else
			start = mid + 1;
		mid = (start + end) / 2;
	}
	std::cout << "0" << "\n";
}



int main()
{
	int x;
    ios_base::sync_with_stdio(0);
	cin.tie(0);

	std::cin >> N;

	for (int i = 0; i < N; i++)
	{
		std::cin >> x;
		arr[i] = x;
	}

	sort(arr, arr+N);

	std::cin >> M;
	for (int i = 0; i < M; i++)
	{
		std::cin >> x;
		BinarySearch(arr, x);
	}

	return 0;
}